dynatrace_health/stackstate_checks/dynatrace_health/dynatrace_health.py

```diff
@@ -0,0 +1,272 @@
+                                           from_time=instance_info.state.last_processed_event_timestamp)
+        new_events = []
+        events_processed = 0
+        event_limit_reached = None
+        try:
+            while events_response:
+                events = events_response.get('events', [])
+                for event in events:
+                    dynatrace_event = DynatraceEvent(event, strict=False)
+                    dynatrace_event.validate()
+                    new_events.append(dynatrace_event)
+                    events_processed += 1
+                    self._check_event_limit_exceeded_condition(instance_info.events_process_limit, events_processed)
+                if events_response.get("nextCursor"):
+                    events_response = self._get_events(dynatrace_client, instance_info.url,
+                                                       cursor=events_response.get("nextCursor"))
+                else:
+                    instance_info.state.last_processed_event_timestamp = events_response.get("to")
+                    events_response = None
+        except EventLimitReachedException as e:
+            instance_info.state.last_processed_event_timestamp = events_response.get("to")
+            event_limit_reached = str(e)
+        return new_events, event_limit_reached
+
+    def _get_events(self, dynatrace_client, url, from_time=None, cursor=None):
+        """
+        Get events from Dynatrace Event API endpoint
+        :param instance_info: object with instance info and its state
+        :param from_time: timestamp from which to collect events
+        :param cursor:
+        :return: Event API endpoint response
+        """
+        params = {}
+        if from_time:
+            params['from'] = from_time
+        if cursor:
+            params['cursor'] = cursor
+        endpoint = url + "/api/v1/events"
+        events = dynatrace_client.get_dynatrace_json_response(endpoint, params)
+        self.log.debug('Got %s events from %s', len(events.get('events', [])), endpoint)
+        return events
+
+    @staticmethod
+    def _check_event_limit_exceeded_condition(events_process_limit, total_event_count):
+        """
+        Raises EventLimitReachedException if number of events between subsequent check runs
+        exceed the `events_process_limit`
+        """
+        if total_event_count >= events_process_limit:
+            raise EventLimitReachedException("Maximum event limit to process is %s but received total %s events"
+                                             % (events_process_limit, total_event_count))
+
+    @staticmethod
+    def _timestamp_to_sts_datetime(dynatrace_event):
+        return datetime.fromtimestamp(dynatrace_event.startTime / 1000).strftime("%b %-d, %Y, %H:%M:%S")
+
+    @staticmethod
+    def _link_to_dynatrace(entity_id, instance_url):
+        entity = dynatrace_entities_cache.get(entity_id)
```

Comment:
`dynatrace_entities_cache` will always be empty as there is no data pushed.

Reply:
Removed entities cache and fixed url generation.